flowpipe/plug.py

```diff
@@ -191,3 +241,154 @@ def connect(self, plug):
         if self not in plug.connections:
             plug.connections.append(self)
             plug.is_dirty = True
+
+    def serialize(self):
+        """Serialize the Plug containing all it's connections."""
+        connections = {}
+        if self.connections:
+            connections[self.connections[0].node.identifier] = self.connections[0].name
+        return {
+            'name': self.name,
+            'value': self.value if not self._sub_plugs else None,
+            'connections': connections,
+            'sub_plugs': {
+                name: sub_plug.serialize()
+                for name, sub_plug in self._sub_plugs.items()
+            }
+        }
+
+
+class SubInputPlug(IPlug):
+    """Held by a parent input plug to form a compound plug."""
+
+    def __init__(self, name, node, parent_plug, value=None):
```

Comment:
Suggestion:
I have seen the assumption of SubPlug names being formated as `{parent_plug.name}.{key}`. If that should be enforced, accept a `key` as a parameter instead of `name` and set the `name` property internally from the parent Plug's name and the SubPlug's key.
In any way, you should also enforce that SubPlug Keys (and Plug names) can not contain a period, by raising a `ValueError` in the respective constructors if the user should attempt that.

Reply:
Very good suggestion, I'll look into that, it should make it more stable, thanks!